### ISSUE TITLE
[BUG] Dataview query for field type Select, Multi and Cycle gives empty result

### ISSUE THREAD

**Issue by OmniNaut** [mdelobelle/metadatamenu#319]
I want to generate a list of Ingredients which go into a meal. Every ingredient has its own note and is tagged with Ingredient in the YAML frontmatter.

Within a Meal note I want to make a list from all the ingredient in the meal
The inline field I use is Ingredients::

When using field types Select, Multi and Cycle, if I use dv.pages("#Ingredient") the returned list is empty. However when I use Link with the same query I will get a list with the proper results.

Using this: 👎🏼
![image](https://user-images.githubusercontent.com/110312864/232380623-80e70ec2-7639-4d56-ada4-123f1fceb480.png)

Results in:
![image](https://user-images.githubusercontent.com/110312864/232380152-4806769b-2c8e-4c49-ac2f-8225edd10c60.png)

Using this:
![image](https://user-images.githubusercontent.com/110312864/232380732-7ddf46ef-1e12-485b-94ab-22fc3c893c14.png)

Results in: 👍🏼
![image](https://user-images.githubusercontent.com/110312864/232380714-da09efae-4a2d-458c-82c3-3b658e15014a.png)

**Comment by Krakor92** [comment]
From what I understand about these types, this is the intended behavior.

`Select`, `Cycle` and `Multi` only accept scalar values, i.e. strings in this context. However, the result of dv.pages("#Ingredients") returns a DataArray (Dataview proxy array implementation) and is therefore not compatible with these field types.

The `Link` field type, on the other hand, expects a DataArray as input, so it behaves as expected.

**Comment by OmniNaut** [comment]
@Krakor92 
Thank you for response.
If what you're saying is true, then I'm really confused. Even in the interface when selecting Select, Cycle or Multi and  with source type selecting "Values returned from a dataview query" the example in the input field says basically the same thing I'm doing.
I admit I'm new to this so maybe I'm missing something here.

![image](https://user-images.githubusercontent.com/110312864/232626109-b4b9ade4-31db-4fc7-9e50-03ce489485fb.png)

**Comment by OmniNaut** [comment]
Ok, so if I modify my query it still can be possible? How would I best approach this?

**Comment by OmniNaut** [comment]
Hi @Krakor92,

I found a partial solution for the problem. I added an extra field inside of a Ingredient note called name. And let the query map against that field. Now I get the desired result. I use the Multi field type.

In the end I completely overhauled my approach, because the approach as described above is too limited. 
I want my ingredients in a table style where I can see the amounts, units and price as well. The old approach did not let me do that.

Anyway, thanks for you help. Appreciate it 👍🏼

I will close this threat as it's not a bug.
